### DISCUSSION TITLE
When does Ulauncher make requests to GitHub?

### DISCUSSION THREAD

**Discussion by daveriedstra** [Ulauncher/Ulauncher#1153] [General]
I recently ran ulauncher from the terminal for the first time in a while and some of the output told me ulauncher couldn't find the versions.json files for some extensions that have long been deprecated:

```
2022-11-15 19:44:47,374 | ERROR | ulauncher.api.server.ExtensionDownloader: download_missing() | GithubExtensionError: Could not find versions.json file using URL "https://raw.githubusercontent.com/compilelife/ulauncher-searchfile/HEAD/versions.json"
2022-11-15 19:44:47,891 | ERROR | ulauncher.api.server.ExtensionDownloader: download_missing() | GithubExtensionError: Could not find versions.json file using URL "https://raw.githubusercontent.com/yegorf1/ulauncher-translator/HEAD/versions.json"
```

I'm assuming there are just some buggy entries in some database and that it's something that won't matter with the eventual rollout of v6. But this looks like ulauncher is making a request to GitHub per extension whenever it starts, which raises a few questions:

1. Is ulauncher actually doing this?
2. When else does ulauncher talk to GitHub? (aside from when explicitly told to, eg, installing & updating extensions)
3. Is ulauncher making other background requests to other servers?
4. Will it continue to make these requests in v6?
5. Can it not? (ie, is there or can there be an option to disable these "background" requests?)

Thanks!

**Comment by friday** [comment]
See #1042

**Reply by daveriedstra** [reply]
Thanks for the reference, that answers question 1 (yes it is) and some of 4 (this particular set of requests is no longer present in v6).

**Reply by friday** [reply]
Ulauncher doesn't spy/track you, but we don't have any policy for when/when not to make requests. We only do it to install, update and check extensions as far as I can remember, but we will extend that to also handle the extension runtime and dependencies (see #273 #721 and #948). In v6 we support GitLab and Codeberg also, so it's not just GitHub any more.

And extensions can make requests also.

**Reply by daveriedstra** [reply]
Thanks for the further clarification. To be clear, I'm not trying to accuse Ulauncher of spying, but I wouldn't put it past GitHub to be analyzing data generated by requests they serve. With something as quotidian as Ulauncher, which gets called dozens of times a day, I want to have a sense of when it's talking to whom. Glad that other forges will be supported in v6! Thanks again!
